ion/tokenizer.go

```diff
@@ -1402,3 +1402,11 @@ func getLongStringCommentsHandler(t *tokenizer, isClob bool) commentHandler {
 	}
 	return t.skipCommentsHandler
 }
+
+func writeCharToStringBuilder(c rune, sb *strings.Builder, isClob bool) {
+	if isClob {
+		sb.WriteByte(byte(c))
```

Comment:
Ahhhhh. This threw me for quite a loop at first, but I think it's correct and I think it's ultimately my fault that it's confusing. :) Explaining what I came up with to check if it matches your understanding:

When `isClob` is true the "string" we're building here is _actually_ the raw unescaped bytes of the clob, which are not necessarily a valid UTF-8 string. (The caller guarantees that `c <= 0xFF` if isClob is true, so the cast to `byte` won't truncate it, but it might be >= 0x80 which WriteRune would UTF8-icize.) We're storing those bytes in a `strings.Builder` to reuse the logic for reading and unescaping _actual_ Ion text strings, but they're not real strings and `textReader.onLob` just grabs the underlying bytes. It'd be more clear to have a separate codepath for clobs that used a `bytes.Buffer` instead of a `strings.Builder` but that would duplicate a lot of code. (Or that's my after-the-fact justification for having done it this way originally, which might be wrong? How gross would it be to refactor?)

If we do keep things this way, it might be worth noting some of this in a comment here. :)

Reply:
> The caller guarantees that c <= 0xFF if isClob is true, so the cast to byte won't truncate it, but it might be >= 0x80 which WriteRune would UTF8-icize

Yes, that sums up why I did this change.

> It'd be more clear to have a separate codepath for clobs that used a bytes.Buffer instead of a strings.Builder but that would duplicate a lot of code. 

It was brought up by @zslayton too in [a previous pull request](https://github.com/amzn/ion-go/pull/130#discussion_r469420291) around clobs, to which I argued code redundancy as well - of course it was before this new situation.

Thoughts @zslayton?

Reply:
The amount of code we need to duplicate does feel pretty substantial; I'd be ok with some explanatory comments. I'll leave it to @Armanbqt's discretion.